### PR TITLE
Friends list and better messages

### PR DESCRIPTION
**Pull request by melektron**
Adds a friend list of players to not report for breaking the beacon.
Adds better discord output for lists of multiple players, showing their position and distance to a triggering event.
Adds more settings in the minecraft.json file.

### CODE REVIEW

File: src/bot.js [hunk 1]
```diff
@@ -56,20 +61,42 @@ const checkBlock = (oldBlock, newBlock) => {
...
+                        // only send mc chat message when no friend was nearby
+                        if (publicReport) sendChatMessage(`${possibleBreakers.join(" or ")} broke ${oldBlock.displayName} at the beacon!`)
+                        
+                        // allways send discord chat messages
+                        sendDiscordMessage(`@break Someone broke ${oldBlock.displayName} ${oldBlock.position} at the beacon!`)
+                        // print more detailed information on discord
+                        let playerPositions = ""
+                        playerPositions = possibleBreakers.reduce((output, currentPlayerKey) => {
+                            const currentPlayer = bot.players[currentPlayerKey]
+                            return output + `     - ${currentPlayer.username} ${friends.includes(currentPlayer.username) ? "(friend)" : ""}: (${currentPlayer.entity.position.x.toFixed(0)}, ${currentPlayer.entity.position.y.toFixed(0)}, ${currentPlayer.entity.position.z.toFixed(0)}\t Δ${currentPlayer.entity.position.distanceTo(oldBlock.position).toFixed(0)})\n`
```
Comment (DarkMetalMouse) [inline src/bot.js]:
```suggestion
                            const pos = currentPlayer.entity.position
                            return output + `     - ${currentPlayer.username} ${friends.includes(currentPlayer.username) ? "(friend)" : ""}: (${pos.x.toFixed(0)}, ${pos.y.toFixed(0)}, ${pos.z.toFixed(0)}\t Δ${pos.distanceTo(oldBlock.position).toFixed(0)})\n`
```

Comment (DarkMetalMouse) [reply src/bot.js]: I think this is less messy

File: src/bot.js [hunk 2]
```diff
@@ -56,20 +61,42 @@ const checkBlock = (oldBlock, newBlock) => {
...
                         lastReportTime = Date.now()
                     }
                 } else {
-                    sendMessage(`@everyone ${oldBlock.displayName} was broken at ${oldBlock.position}, but I don't know who did it :(`)
+                    // if no players could be detected close enough, still inform discord about the changes and all visible players
+                    let playerPositions = ""
+                    playerPositions = Object.keys(bot.players).reduce((output, currentPlayerKey) => {
+                        const currentPlayer = bot.players[currentPlayerKey]
+                        if (currentPlayer.entity == null) return output    // ignore players that are out of view distance
+                        return output + `     - ${currentPlayer.username} ${friends.includes(currentPlayer.username) ? "(friend)" : ""}: (${currentPlayer.entity.position.x.toFixed(0)}, ${currentPlayer.entity.position.y.toFixed(0)}, ${currentPlayer.entity.position.z.toFixed(0)}\t Δ${currentPlayer.entity.position.distanceTo(oldBlock.position).toFixed(0)})\n`
```
Comment (DarkMetalMouse) [inline src/bot.js]: Same here, extract pos to a different variable

File: src/minecraft_old.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "light_blue_stained_glass",
+        "white_stained_glass"
+    ],
+    "friends": [
+        "DarkMetalMouse",
+        "ElektronMC"
+    ],
+    "sendPublicMCMessages": true,
+    "playerAssosiationRadius": 7
+}
```
Comment (DarkMetalMouse) [inline src/minecraft_old.json]: Why is this file here?

Comment (melektron) [reply src/minecraft_old.json]: This file contains the coordinates on the real server and the other one contains test coordinates

File: src/bot.js (hunk 1 above)
Comment (melektron) [reply src/bot.js]: That's a good idea, will implement
